core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -78,9 +76,6 @@ public class RemoteStorageManager implements org.apache.kafka.server.log.remote.
 
     private final Executor executor = new ForkJoinPool();
 
-    private ObjectFetcher fetcher;
-    private ObjectUploader uploader;
-    private ObjectDeleter deleter;
```

Comment:
From our discussions, we explicitly wanted these to be separate, primarily to have separate S3 clients for different purposes.

Reply:
Should then this be documented? I vaguely remember that but it's not obvious

Reply:
Adjusting the PR accordingly

Reply:
However I still think that we then need to create separate clients inside s3 implementation instead.